In [2]:
import pandas as pd
predictions = pd.read_csv("../outputs/segmentation/segmentation_clients.csv")
print(predictions[predictions["Segment"] == "Nouveaux / à activer"])

     Customer_ID          Name  Age Gender     Location  Churn  Recency  \
633         2634  Robert Reese   44   Male  San Antonio      0       30   

     Frequency  Monetary  Avg_Basket  R_score  F_score  M_score  RFM_score  \
633          9   2984.73      331.64        4        2        2          8   

     Cluster               Segment  \
633        2  Nouveaux / à activer   

                                        Recommandation  
633  Onboarding, offre de bienvenue, incitation 2èm...  


### Vérification — client "Nouveaux / à activer" (seul cas)

Chargement de `segmentation_clients.csv` et isolation du seul client
classé "Nouveaux / à activer" (Customer_ID 2634). Profil confirmé
cohérent avec la règle de `label_from_rfm` (R_score=4, F_score=2,
M_score=2 : achat récent mais peu fréquent et peu dépensier — typique
d'un client encore en phase d'activation). Pas un bug du pipeline,
juste un segment naturellement rare dans ce dataset synthétique.

In [3]:
predictions.groupby("Segment")["Churn"].agg(["mean", "count"])

,mean,count
Segment,,
Clients fidèles,0.000000,118
Nouveaux / à activer,0.000000,1
Occasionnels,0.128405,257
Perdus,1.000000,365
VIP / Champions,0.000000,246
À risque,0.769231,13


### Interprétation — taux de churn par segment (validation croisée)

| Segment | Taux de churn | Clients |
|---|---|---|
| Perdus | 100% | 365 |
| À risque | 76.9% | 13 |
| Occasionnels | 12.8% | 257 |
| VIP / Champions | 0% | 246 |
| Clients fidèles | 0% | 118 |
| Nouveaux / à activer | 0% | 1 |

**Cohérence quasi parfaite avec les noms des segments** : Perdus = 100%
churn, À risque = 77% churn, VIP/Fidèles = 0% churn. Le RFM + clustering
capture bien le signal de churn, malgré la rareté de certains segments
(À risque, Nouveaux) — confirme que la répartition déséquilibrée est
une caractéristique du dataset, pas un défaut du pipeline.